javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -0,0 +1,1761 @@
+        val q = this.getUnscaledValue.divide(div)
+        // To chose  10^newScale  to get a quotient with at least 'mc.precision()' digits
+        val ns = Math.min(diffScale, Math.max(mcPrecision - quotPrecision + 1, 0))
+        (q.multiply(powerOf10(ns)), ns)
+      } else {
+        /* To calculate the minimum power of ten, such that the quotient
+         * (u1 * 10^exp) / u2   has at least 'mc.precision()' digits. */
+        val exp = Math.min(-diffScale, Math.max(mcPrecision.toLong - diffPrecision, 0))
+        val mult = this.getUnscaledValue.multiply(powerOf10(exp))
+        val qr: QuotAndRem[BigInteger] = mult.divideAndRemainder(divisor.getUnscaledValue)
+        val ns = diffScale + exp // To fix the scale
+        val exp2 = -ns // The remaining power of ten
+        // If after division there is a remainder...
+        if ((qr.rem.signum() != 0) && (exp2 > 0)) {
+          val bi = new BigDecimal(qr.rem)
+          val compRemDiv0 = bi.precision() + exp2 - divisor.precision()
+          val compRemDiv = {
+            if (compRemDiv0 == 0) {
+              val bi = qr.rem.multiply(powerOf10(exp2))
+              val rem = bi.divide(divisor.getUnscaledValue)
+              Math.abs(rem.signum())
+            } else {
+              compRemDiv0
+            }
+          }
+          if (compRemDiv > 0)
+            throw new ArithmeticException("Division impossible")
+        }
+        (qr.quot, ns)
+      }
+    }
+
+    // Fast return if the quotient is zero
+    if (quot.signum() == 0)
+      return zeroScaledBy(diffScale)
+
+    val integralValue = new BigDecimal(quot)
+
+    // To strip trailing zeros until the specified precision is reached
+    @inline
+    @tailrec
+    def loop(i: Int, ns: Long, q: BigInteger, prec: Int): (Long, BigInteger, Int) = {
+      if(!q.testBit(0)) {
+        val qr: QuotAndRem[BigInteger] = q.divideAndRemainder(BigTenPows(i))
+        val cond1 = (qr.rem.signum() == 0) &&
+            ((prec - i >= mcPrecision) || (ns - i >= diffScale))
+
+        if (cond1) loop(if (i < lastPow) i + 1 else i, ns - i, qr.quot, prec - i)
+        else if (i != 1) loop(1, ns, q, prec)
+        else (ns, q, prec)
+      } else {
+        (ns, q, prec)
+      }
+    }
+
+    val (finalScale, strippedBI, resultPrecision) =
+      loop(1, newScale, quot, integralValue.precision())
+
+    // To check if the result fit in 'mc.precision()' digits
+    if (resultPrecision > mcPrecision)
+      throw new ArithmeticException("Division impossible")
+
+    integralValue._scale = safeLongToInt(finalScale)
+    integralValue.setUnscaledValue(strippedBI)
+    integralValue
+  }
+
+  def remainder(divisor: BigDecimal): BigDecimal =
+    divideAndRemainderImpl(divisor).rem
+
+  def remainder(divisor: BigDecimal, mc: MathContext): BigDecimal =
+    divideAndRemainderImpl(divisor, mc).rem
+
+  @inline
+  def divideAndRemainder(divisor: BigDecimal): Array[BigDecimal] =
+    divideAndRemainderImpl(divisor)
```

Comment:
Don't rely on an implicit conversion for this. Bring the content of the implicit conversion right inside this method instead.
Otherwise, we can't be sure whether the implicit conv. is not also used in places we don't want to.

Also, don't mark this method as `@inline`. In this case it is not worth it, because Arrays are never stack-allocated anyway.